Source/WebCore/platform/graphics/Image.h

```diff
@@ -203,7 +208,10 @@ class Image : public RefCounted<Image> {
 #if PLATFORM(WIN)
     virtual void drawFrameMatchingSourceSize(GraphicsContext&, const FloatRect& dstRect, const IntSize& srcSize, CompositeOperator) { }
 #endif
+    virtual bool canDrawViewport(GraphicsContext&) const { return false; }
```

Comment:
I think using "viewport" here is confusing (it assumes knowledge of some internal behavior of Image). It's more about "canUseCachedRepresentation()" or something.

Reply:
I will think about a better name.